src/main/java/com/solr/clientwrapper/domain/service/SolrCoreService.java

```diff
@@ -0,0 +1,226 @@
+package com.solr.clientwrapper.domain.service;
+
+import com.solr.clientwrapper.domain.dto.solr.SolrResponseDTO;
+import com.solr.clientwrapper.domain.port.api.SolrCoreServicePort;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.client.solrj.request.CoreAdminRequest;
+import org.apache.solr.client.solrj.response.CoreAdminResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.io.IOException;
+
+@Service
+@Transactional
+public class SolrCoreService implements SolrCoreServicePort {
+
+    HttpSolrClient solrClient = new HttpSolrClient.Builder("http://localhost:8983/solr").build();
+
+    @Override
+    public SolrResponseDTO createCore(String coreName) {
+        //HttpSolrClient solrClient = new HttpSolrClient.Builder("http://localhost:8983/solr").build();
+        CoreAdminRequest.Create request = new CoreAdminRequest.Create();
+        request.setCoreName(coreName);
+        request.setInstanceDir(coreName);
+        request.setConfigSet("_default");
+        request.setDataDir("data");
+
+        SolrResponseDTO solrResponseDTO=new SolrResponseDTO();
+        solrResponseDTO.setCoreName(coreName);
+
+        try {
+            CoreAdminResponse coreAdminResponse =request.process(solrClient);
+
+            solrResponseDTO.setStatusCode(200);
+            solrResponseDTO.setMessage("Successfully created Solr Core: "+coreName);
+
+        } catch (SolrServerException e) {
+            e.printStackTrace();
+
+            solrResponseDTO.setStatusCode(400);
+            solrResponseDTO.setMessage("Unable to create Solr Core: "+coreName+". SolrServerException.");
+
+        } catch (IOException e) {
```

Comment:
https://www.journaldev.com/629/java-catch-multiple-exceptions-rethrow-exception
catch multiple exceptions in one to reduce redundant code

Reply:
Resolved.